### PR TITLE
Add default github workflow

### PR DESCRIPTION
**Pull request by ereslibre**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      # string substitutions on this template, so we can support
+      # targets with regular dashes that cargo replaces with
+      # underscore.
+      - name: Copy Wasm module to an expected file name
+        run: |
+          cp "target/wasm32-unknown-unknown/release/$(echo "{{project-name}}.wasm" | sed 's/-/_/')" target/wasm32-unknown-unknown/release/{{project-name}}.wasm
+      - name: Publish Wasm policy artifact to OCI registry with the 'latest' tag
+        if: {{ "${{ startsWith(github.ref, 'refs/heads/') }}" }}
+        run: |
+          ./linux-amd64-wasm-to-oci push target/wasm32-unknown-unknown/release/{{project-name}}.wasm ghcr.io/kubewarden/policies/{{project-name}}:latest
```
Comment (ereslibre) [inline .github/workflows/release.yml:53]: We should probably not leave kubewarden here... wdyt? The only alternative I see is to improve the workflow to use the default organization where the project started out of the template lives currently. That might need some extra steps on the GitHub workflow though. [1]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      - name: Publish Wasm policy artifact to OCI registry with the version tag and 'latest'
+        if: {{ "${{ startsWith(github.ref, 'refs/tags/') }}" }}
+        run: |
+          export OCI_TAG=$(echo $GITHUB_REF | sed -e "s|refs/tags/||")
+          ./linux-amd64-wasm-to-oci push target/wasm32-unknown-unknown/release/{{project-name}}.wasm ghcr.io/kubewarden/policies/{{project-name}}:${OCI_TAG}
```
Comment (ereslibre) [inline .github/workflows/release.yml:58]: [1]

File: .github/workflows/release.yml (hunk 1 above)
Comment (flavio) [reply .github/workflows/release.yml:53]: What about making that an GitHub action variable? We could make the default value something like `<TODO replace me with your target repository>`

File: .github/workflows/release.yml (hunk 2 above)
Comment (flavio) [reply .github/workflows/release.yml:58]: Same here

**Review by flavio — CHANGES_REQUESTED** [review]
LGTM, I would just do the fixes I mentioned.

Finally, I would recommend to replace the GH action of one of our policies with what would be generated by `cargo generate`. That way we would be 100% sure the action is working as intended.

**Comment by ereslibre** [comment]
> Finally, I would recommend to replace the GH action of one of our policies with what would be generated by cargo generate. That way we would be 100% sure the action is working as intended.

I am fine doing it this way, but I would rather add yet another pipeline within this project that does that test, both after push/PR and in a scheduled way.

**Comment by ereslibre** [comment]
Of course, assuming we have a way to not include the "test the template" workflow inside the target project generated by cargo generate itself...

**Comment by ereslibre** [comment]
Just tested on a clean repo of my own. Everything worked fine and carg-generate now asks for a couple more placeholders with default values: registry (`ghcr.io`) by default and prefix of the image (`kubewarden/policies` by default).

**Pull request merged**